File: FUNC_LIB/UBC_LP_Wrappers/Lp_MAG3D/MAG_functions/read_input.m
function [home_dir,UBC_dir,iter_start,chifact,pvec,qvec,lvec,cool_beta,iter_max] = read_input(input_file)
% [home_dir,UBC,dir,iter_start,chifact,pvec,qvec,lvec,cool_beta,iter_max] = read_input(input_file);
% Read input file for compact inversion
% 
% INPUT//
% File and directory
% 
% OUTPUT//
% home_dir  : Directory on local drive containing *.mtx, *.msh, *.topo files
% UBC_dir   : Directory for the uncontrained UBC results (models & *.log)
% iter_start: Starting UBC model #
% chifact   : Target chi-factor
% pvec      : Value(s) for the norm on gradient in range [0 2]
% qvec      : Value(s) for the norm on model in range [0 2]
% lvec      : Value(s) for the scaling between gradient and model norm in range [0 2]
% // If multiple values are used for any of p, q or l, the algorithm will
% invert for every combinaison. Results will be stored in seperate folders.
% cool_beta : Cooling schedule for the trade-off parameter (default 0.5)
% iter_max  : Maximum number of iteration

fid=fopen(input_file,'rt');

% Go through the input file and extract parameters
    
% First line: home_directory 
line=fgets(fid);
delm=regexp(line,'!');
home_dir = strtrim(line(1:delm-1));

% Second line: UBC_results directory 
line=fgets(fid);
delm=regexp(line,'!');
UBC_dir = strtrim(line(1:delm-1));

% Third line: Starting UBC model 
line=fgets(fid);
delm=regexp(line,'!');
iter_start = strtrim(line(1:delm-1));

% Fourth line: Target chi factor 
line=fgets(fid);
delm=regexp(line,'!');
chifact = str2num(strtrim(line(1:delm-1)));

% Fifth line: lp norm on gradient 
line=fgets(fid);
delm=regexp(line,'!');
pvec = str2num(strtrim(line(1:delm-1)));

% Sixth line: lq norm on gradient 
line=fgets(fid);
delm=regexp(line,'!');
qvec = str2num(strtrim(line(1:delm-1)));

% Seventh line: lambda scaling 
line=fgets(fid);
delm=regexp(line,'!');
lvec = str2num(strtrim(line(1:delm-1)));

% Eighth line: lp norm on gradient 
line=fgets(fid);
delm=regexp(line,'!');
cool_beta = str2num(strtrim(line(1:delm-1)));

% Nineth line: lp norm on gradient 
line=fgets(fid);
delm=regexp(line,'!');
iter_max = str2num(strtrim(line(1:delm-1)));

File: FUNC_LIB/UBC_LP_Wrappers/Python_scripts/Lp_GRAV3D/src/read_input.py
'''
Created on Jul 17, 2013

@author: dominiquef
'''
def read_input(input_file):
    """Read input files for the lp_norm script"""
    import os
    import re
    from numpy import asarray
    
    os.system('pwd')
    fid = open(input_file,'r')
    
    line = fid.readline()
    l_input  = line.split('!')
    Home_dir = l_input[0].rstrip()
    
    line = fid.readline()
    l_input  = line.split('!')
    UBC_dir = l_input[0].rstrip()
    
    line = fid.readline()
    l_input = line.split('!') 
    iter_start = int(l_input[0])
    
    line = fid.readline()
    l_input = line.split('!') 
    chifact = float(l_input[0])
    
    line = fid.readline()
    l_input = line.split('!')
    pp = l_input[0].split(' ')
    p = asarray(pp)
    
    line = fid.readline()
    l_input = line.split('!')
    qq = l_input[0].split(' ')
    q = asarray(qq)
    
    line = fid.readline()
    l_input = line.split('!')
    ll = l_input[0].split(' ')
    l = asarray(ll)
    
    line = fid.readline()
    l_input = line.split('!') 
    cool_beta = float(l_input[0])
    
    line = fid.readline()
    l_input = line.split('!') 
    iter_max = float(l_input[0])
      
    return Home_dir, UBC_dir, iter_start, chifact, p, q, l, cool_beta, iter_max